### DISCUSSION TITLE
API endpoint for changes the current configuration does not update paths

### DISCUSSION THREAD

**Discussion by ikcehcus** [bluenviron/mediamtx#530] [Questions]
Hi,
Does the endpoint for update the current config should update paths? Because currently, something does not work.
I can change single property, but the paths section is never updated.
I have a case, that every day I must remove hundreds path and add new. I am afraid that send a single request to delete and add will take too much time.
Regards

**Comment by aler9** [comment]
Hello, this is a design choice, it's better not to provide two ways to do the same thing.
Anyway, it's not difficult to send even 200 path creation requests at a time, i tested with 500 requests at it took less than 3 secs. Some time ago there was a bug that sometimes prevented path creation with the API (#542) but it's fixed in v0.17.3.

**Reply by ikcehcus** [reply]
but I have a problem with multiple single requests
something happens stranger things, all requests is 200 but logs look like this:

2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-main] destroyed
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-sub] destroyed
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-sub] created

should look like this:

2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-main] destroyed
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-sub] destroyed
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
**2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-main] created**
2021/09/23 12:31:55 I [0/2] reloading configuration (API request)
2021/09/23 12:31:55 I [0/2] [path 9a23b428-4f23-4e86-b512-6ebbc38ccbf6-1-sub] created

Configuration is reloaded but no info in logs about path created.

**Reply by ikcehcus** [reply]
probably problem exists when I delete paths before create

**Reply by aler9** [reply]
If you delete a path before creating then it's not a bug - otherwise, if you think it's a bug, please provide a way to replicate the issue.

[a comment by an automated account is not shown]
